lib/puppet/provider/service/systemd.rb

```diff
@@ -32,7 +32,14 @@ def disable
   end
 
   def get_start_link_count
-    Dir.glob("/etc/rc*.d/S??#{@resource[:name]}").length
```

Comment:
Yikes. Good sleuthing. That makes sense (in a Debian-sort of way :)). I wonder if dropping some of your note here in as a comment in the code will help future-me when I squint at this again in six months.